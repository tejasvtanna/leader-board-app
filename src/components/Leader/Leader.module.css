.leader {
  margin: 10px 20px;
  border: lightgrey solid 1px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 2rem;
  padding: 0.4rem;
  border-radius: 5px;
  background-color: #efefef;
}

.name {
  width: 70%;
  font-weight: bold;
}

.score {
  color: #749adc;
  text-shadow: 1px 1px #749adc;
  font-weight: bold;
}

.plusMinus {
  width: 2rem;
  border-color: lightgrey;
}

.plusMinus:hover {
  cursor: pointer;
}
