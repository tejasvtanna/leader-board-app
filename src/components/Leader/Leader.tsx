import React from 'react'
import { ILeader } from 'interfaces/Leader.interface'
import classes from './Leader.module.css'

interface Props {
  leader: ILeader
  onIncrement: (id: string) => {}
  onDecrement: (id: string, marks: number) => {}
}

const Leader = ({ leader, onIncrement, onDecrement }: Props) => {
  return (
    <div className={classes.leader}>
      <div className={classes.name}>{leader.name}</div>
      <div className={classes.score} data-testid="points">
        {leader.points}
      </div>
      <button className={classes.plusMinus} onClick={() => onIncrement(leader.id)} data-testid="btnIncrement">
        +
      </button>
      <button
        className={classes.plusMinus}
        onClick={() => onDecrement(leader.id, leader.points)}
        data-testid="btnDecrement">
        -
      </button>
    </div>
  )
}

export default Leader
