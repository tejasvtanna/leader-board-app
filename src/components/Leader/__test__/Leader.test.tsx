import React from 'react'
import Leader from '../Leader'
import { fireEvent, render, screen, cleanup, waitFor } from '@testing-library/react'
import { ILeader } from 'interfaces/Leader.interface'
const renderer = require('react-test-renderer')

afterEach(cleanup)

describe('Leader', () => {
  test('snapshot testing', () => {
    const leader: ILeader = { id: '1', name: 'Tejas', points: 1 }
    const onIncrement = jest.fn
    const onDecrement = jest.fn

    const component = renderer.create(<Leader leader={leader} onIncrement={onIncrement} onDecrement={onDecrement} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('calls increment/decrement functions', () => {
    const leader: ILeader = { id: '1', name: 'Tejas', points: 1 }
    const onIncrement = jest.fn()
    const onDecrement = jest.fn()

    render(<Leader leader={leader} onIncrement={onIncrement} onDecrement={onDecrement} />)

    fireEvent.click(screen.getByTestId('btnIncrement'))
    expect(onIncrement).toHaveBeenCalled()

    fireEvent.click(screen.getByTestId('btnDecrement'))
    expect(onDecrement).toHaveBeenCalled()
  })
})
