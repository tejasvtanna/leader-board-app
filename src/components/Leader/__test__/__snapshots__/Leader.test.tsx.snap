// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leader snapshot testing 1`] = `
<div
  className="leader"
>
  <div
    className="name"
  >
    Tejas
  </div>
  <div
    className="score"
    data-testid="points"
  >
    1
  </div>
  <button
    className="plusMinus"
    data-testid="btnIncrement"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="plusMinus"
    data-testid="btnDecrement"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
